crates/rust-analyzer/src/global_state.rs

```diff
@@ -365,6 +412,54 @@ impl GlobalState {
 
         self.analysis_host.apply_change(change);
 
+        let config_change = {
```

Comment:
We should skip all of the green diff here if `ratoml_text_map.is_empty() && user_config_file.is_none() && root_path_ra_tom.is_none()` (otherwise we unnecessarily replace the config with an equal copy/do unnecessary work). This function we are in runs on the main thread and also cancels all outstanding tasks, so as long as this function runs it will block the entire main loop from progressing. So we need ot be careful with the amount of work we do here.